File: src/main/java/net/guizhanss/ultimategenerators2/core/recipes/SingleOutputMachineRecipe.java
package net.guizhanss.ultimategenerators2.core.recipes;

import org.bukkit.inventory.ItemStack;

/**
 * A {@link MachineRecipe} but only has one output.
 */
public class SingleOutputMachineRecipe extends MachineRecipe {
    public SingleOutputMachineRecipe(int ticks, ItemStack[] input, ItemStack output) {
        super(ticks, input, new ItemStack[] {output});
    }
}

File: src/main/kotlin/net/guizhanss/ultimategenerators2/core/recipes/SingleOutputMachineRecipe.kt
package net.guizhanss.ultimategenerators2.core.recipes

import org.bukkit.inventory.ItemStack

/**
 * A [MachineRecipe] but only has one output.
 */
open class SingleOutputMachineRecipe(ticks: Int, input: Array<ItemStack>, output: ItemStack) :
    MachineRecipe(ticks, input, arrayOf(output))
